### PR TITLE
Enchanting & Tradeskill vendor rework

### LINKED ISSUE #920: Enchanting suppliers/trade suppliers should sell Mystical Powder
**Describe the bug**
Mystical Powder (6216) is required to craft enchants. However, it is currently not sold by any vendor and there is no other way for the player to obtain it. This renders the enchanting profession unusable.

**Steps to reproduce**
1. Learn enchanting
2. Check any tradeskill or enchanting related vendor, anywhere in the world.
3. Be very sad.

**Expected behavior**
Mystical Powder should be readily available from any enchanting or profession related vendor just like any other common trade skill resource. Given than you can learn enchanting very early in the starting areas of each race one can assume *every* related vendor indeed sold this item.

**Sources**
According to [this source](https://crawler.thealphaproject.eu/mnt/crawler/media/Text/Various/alpha-tester-report-march-18-2004.txt), the undead push - which is 0.5.3 - included the enchanting profession ready to use:

> This was added to the game in the undead push, and **that's when I tried it**. Magical effects can be added to various pieces of armor – the effect and the target depend on the recipe – using herbs, magical dust and copper and silver rods. In order to keep things balanced, permanent enchantments use a lot of herbs, and most of the skilling up recipes are temporary imbuing recipes.

It's impossible to "try it" if a core ingredient is missing from the game.

### PR DESCRIPTION
**Pull request by mindphluxnet**
Closes #920 

- Renames enchanting vendors to "Enchanter"/"Enchantress"
- removes "Bundle of Wood" from enchanting vendors; it's not enchanting related
- adds enchanting powder items to enchanting vendors
- adds ordered vendor template for tradeskill vendors
- assigns the new template to all tradeskill vendors

### CODE REVIEW

File: etc/databases/world/updates/updates.sql [hunk 1]
```diff
@@ -30510,5 +30510,72 @@ begin not atomic
         insert into applied_updates values ('080320231');
     end if;
 
+    -- 09/03/2023 1
+    if (select count(*) from applied_updates where id = '090320231') = 0 then
+        -- Set "Enchanter" subname for male Enchanting Vendors.
+        update creature_template set subname = 'Enchanter' where entry in(4228, 4617, 5758);
+        -- And "Enchantress" for females.
+        update creature_template set subname = 'Enchantress' where entry in(1318, 3012, 3346, 4228, 5158, 5757);
```
Comment (GrenderG) [inline etc/databases/world/updates/updates.sql:30518]: Do we have any evidence of this?

Comment (mindphluxnet) [reply etc/databases/world/updates/updates.sql:30518]: No but it's blizzlike, isn't it?

File: etc/databases/world/updates/updates.sql [hunk 2]
```diff
@@ -30510,5 +30510,72 @@ begin not atomic
...
+        insert into npc_vendor_template (entry, item, slot) values (66, 2928, 29);
+        insert into npc_vendor_template (entry, item, slot) values (66, 8925, 30);
+        insert into npc_vendor_template (entry, item, slot) values (66, 4400, 31);
+        insert into npc_vendor_template (entry, item, slot) values (66, 6532, 32);
+        insert into npc_vendor_template (entry, item, slot) values (66, 3372, 33);
+        insert into npc_vendor_template (entry, item, slot) values (66, 4399, 34);
+        insert into npc_vendor_template (entry, item, slot) values (66, 6530, 35);
+
+        -- Assign tradeskill vendors the new vendor template.
+        update creature_template set vendor_id = 66 where entry in(2118, 5163, 4561, 1250, 1286, 5100, 3168, 843, 4897, 4877, 4229, 3614,
```
[the remark this answers is not in the document]
Comment (mindphluxnet) [reply etc/databases/world/updates/updates.sql:30572]: Done

File: etc/databases/world/updates/updates.sql (hunk 1 above)
Comment (mindphluxnet) [reply etc/databases/world/updates/updates.sql:30518]: There's one vendor already having that subname by the way:

https://db.thealphaproject.eu/index.php?action=show_creature&id=3606&filter=enchantress&sort_order=name&pos=1&max=1

File: etc/databases/world/updates/updates.sql [hunk 3]
```diff
@@ -30510,5 +30510,76 @@ begin not atomic
...
+         3499, 3187, 3482, 4194);
```
Comment (GrenderG) [inline etc/databases/world/updates/updates.sql:30573]: NPC entry `66` is missing here I think.

File: etc/databases/world/updates/updates.sql [hunk 4]
```diff
@@ -30510,5 +30510,76 @@ begin not atomic
         insert into applied_updates values ('080320231');
     end if;
 
+    -- 09/03/2023 1
+    if (select count(*) from applied_updates where id = '090320231') = 0 then
+        -- Set "Enchanter" subname for male Enchanting Vendors.
+        update creature_template set subname = 'Enchanter' where entry in(4228, 4617, 5758);
+        -- And "Enchantress" for females.
+        update creature_template set subname = 'Enchantress' where entry in(1318, 3012, 3346, 4228, 5158, 5757);
```
Comment (GrenderG) [inline etc/databases/world/updates/updates.sql:30518]: Just noticed `4228 | Vaean` appears in both `Enchanter` and `Enchantress` queries. I assume he needs to be in the first query only.

Comment (mindphluxnet) [reply etc/databases/world/updates/updates.sql:30518]: Fixed

**Pull request merged**